ddtrace/contrib/pylons/patch.py

```diff
@@ -14,27 +17,30 @@ def patch():
         return
 
     setattr(pylons.wsgiapp, '_datadog_patch', True)
-    wrapt.wrap_function_wrapper('pylons.wsgiapp', 'PylonsApp.__init__', traced_init)
+    setattr(pylons.wsgiapp, 'PylonsApp', TracedPylonsApp)
 
 
 def unpatch():
     """Disable Pylons tracing"""
-    if not getattr(pylons.wsgiapp, '__datadog_patch', False):
+    if not getattr(pylons.wsgiapp, '_datadog_patch', False):
         return
-    setattr(pylons.wsgiapp, '__datadog_patch', False)
+    setattr(pylons.wsgiapp, '_datadog_patch', False)
+
+    setattr(pylons.wsgiapp, 'PylonsApp', _PylonsApp)
 
-    _u(pylons.wsgiapp.PylonsApp, '__init__')
 
+class TracedPylonsApp(wrapt.ObjectProxy):
+    def __init__(self, *args, **kwargs):
```

Comment:
Is it possible for us to add a deprecation warning for anyone who is manually using this?

Reply:
We could, but a problem with Pylons is that our auto-patched middleware is unable to cover user installed middleware. (And I don't see a good way to address this). So I think it's fine to continue to support it this way.

Reply:
well, I am thinking:

```python
# We want this
from ddtrace import patch_all; patch_all(pylons=True)

# Show a deprecation warning
from ddtrace.contrib.pylons import TracedPylonsApp
app TracedPylonsApp()
```

Reply:
Yeah, the problem is that the manual approach provides more insight into the actual working of pylons. I'm not sure we want to deprecate it until our auto-patching is equivalent or better.

Reply:
:+1: